pkg/issuercerts/issuercerts.go

```diff
@@ -0,0 +1,125 @@
+
+	secret, err := api.CoreV1().Secrets(controlPlaneNamespace).Get(k8s.IdentityIssuerSecretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	crt, ok := secret.Data[k8s.IdentityIssuerCrtName]
+	if !ok {
+		return nil, fmt.Errorf(кeyMissingError, k8s.IdentityIssuerCrtName, "issuer certificate", consts.IdentityIssuerSecretName, false)
+	}
+
+	key, ok := secret.Data[k8s.IdentityIssuerKeyName]
+	if !ok {
+		return nil, fmt.Errorf(кeyMissingError, k8s.IdentityIssuerKeyName, "issuer key", consts.IdentityIssuerSecretName, true)
+	}
+
+	return &IssuerCertData{trustAnchors, string(crt), string(key)}, nil
+}
+
+// FetchExternalIssuerData fetches the issuer data from the linkerd-identitiy-issuer secrets (used for kubernetes.io/tls schemed secrets)
+func FetchExternalIssuerData(api *k8s.KubernetesAPI, controlPlaneNamespace string) (*IssuerCertData, error) {
+	secret, err := api.CoreV1().Secrets(controlPlaneNamespace).Get(k8s.IdentityIssuerSecretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	anchors, ok := secret.Data[consts.IdentityIssuerTrustAnchorsNameExternal]
+	if !ok {
+		return nil, fmt.Errorf(кeyMissingError, consts.IdentityIssuerTrustAnchorsNameExternal, "trust anchors", consts.IdentityIssuerSecretName, true)
+	}
+
+	crt, ok := secret.Data[corev1.TLSCertKey]
+	if !ok {
+		return nil, fmt.Errorf(кeyMissingError, corev1.TLSCertKey, "issuer certificate", consts.IdentityIssuerSecretName, true)
+	}
+
+	key, ok := secret.Data[corev1.TLSPrivateKeyKey]
+	if !ok {
+		return nil, fmt.Errorf(кeyMissingError, corev1.TLSPrivateKeyKey, "issuer key", consts.IdentityIssuerSecretName, true)
+	}
+
+	return &IssuerCertData{string(anchors), string(crt), string(key)}, nil
+}
+
+// LoadIssuerDataFromFiles loads the issuer data from file stored on disk
+func LoadIssuerDataFromFiles(keyPEMFile, crtPEMFile, trustPEMFile string) (*IssuerCertData, error) {
+	key, err := ioutil.ReadFile(keyPEMFile)
+	if err != nil {
+		return nil, err
+	}
+
+	crt, err := ioutil.ReadFile(crtPEMFile)
+	if err != nil {
+		return nil, err
+	}
+
+	anchors, err := ioutil.ReadFile(trustPEMFile)
+	if err != nil {
+		return nil, err
+	}
+
+	return &IssuerCertData{string(anchors), string(crt), string(key)}, nil
+}
+
+// BuildCreds builds and validates the creds out of the data in IssuerCertData
+func (ic *IssuerCertData) BuildCreds(dnsName string) (*tls.Cred, error) {
+
+	creds, err := tls.ValidateAndCreateCreds(ic.IssuerCrt, ic.IssuerKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read CA: %s", err)
+	}
+
+	algo := creds.Certificate.PublicKeyAlgorithm
+	starts := creds.Certificate.NotBefore
+	ends := creds.Certificate.NotAfter
+
+	if algo != x509.ECDSA {
+		return nil, fmt.Errorf("the required public key algorithm is %s, instead %s was used", x509.ECDSA, algo)
+	}
+
+	if starts.After(time.Now()) {
+		return nil, fmt.Errorf("certificate not valid before: %s", starts.Format(time.RFC3339))
+	}
+
+	if ends.Before(time.Now()) {
```

Comment:
AFAIU, we're checking the credentials aginst the linkerd CLI's clock. Is that sufficient?

Reply:
Might be wrong here, but assuming a --pre check has passed and concluded that there is [no clock skew](https://github.com/linkerd/linkerd2/blob/4b6254b52ef84c42fb72b9473fe9cd7ae0721f8b/pkg/healthcheck/healthcheck.go#L1416), wouldnt it be sufficient?